codex/slots/sampler/utils.nim

```diff
@@ -80,7 +69,7 @@ func cellIndices*(
   numCells: Natural, nSamples: Natural): seq[Natural] =
 
   var indices: seq[Natural]
-  while (indices.len < nSamples):
-    let idx = cellIndex(entropy, slotRoot, numCells, indices.len + 1)
-    indices.add(idx.Natural)
+  for i in 1..nSamples:
+    indices.add(cellIndex(entropy, slotRoot, numCells, i))
```

Comment:
This changes the logic by passing a different `counter` to `cellIndex`, this looks like a breaking change?

Reply:
Ah, you're right. The previous logic is quite confusing indeed.